pkg/helper/secretutils.go

```diff
@@ -97,6 +97,24 @@ func NewSecretSource(client client.Client, namespace string) *SecretSource {
 	}
 }
 
+// GetField returns the value of the secret field named fieldName in
+// the Secret named secretName.
+// If the field fieldName does not exist in the secret secretName or the secret
+// secretName does not exists a nil string is returned.
+// If there's an error processing the SecretSource an error is returned.
+func (s *SecretSource) GetField(secretName, fieldName string) (*string, error) {
```

Comment:
Try to re-use `func (s *SecretSource) fieldReader(secretName, fieldName string, secretRequired, fieldRequired bool, def string) (string, error)` instead of re implementing. 


Your use case is about having different logic when the secret (or the secret-key) does not exist and when the value exists. For that regard, call `RequiredFieldValueFromRequiredSecret` method and check for errors about a) secret not found and b) field value not found. If any of those errors happen, you know you shouldn't add the env var. If any other error type is found, then report back and done. It would require raising some known specific type of error [here](https://github.com/3scale/3scale-operator/blob/f504a03941ec566e301a146132b3f02575caee62/pkg/helper/secretutils.go#L126) in order to check later, like
```
if errors.IsNotFound(err) || helper.IsSecretKeyNotFound(err) {
// do not add the env var
}
```


An alternative implementation is using the empty string as the default value `""`. It's valid approach because the empty string is not a valid FROM field value, hence if you get the default value, you know you should not set the env var. With this approach, if the customer sets the empty value, the value will not be propagated to system. This is the main difference with the previous approach.

Reply:
I prefer the later proposed approach.

I've updated the PR with a new commit to change the behavior. If it looks ok I'll squash it.
The new behavior is:
* We only set System's environment variable when the from address secret field is set AND it is set to a non-empty value.
* By default, we explicitly set the from address field into the 'smtp' secret with the vale `""` to be explicit and consistent with the other fields. From the point of view of the user `""` means "use the default value" as with the other fields and the user can explicitly set it too with that meaning. An upgrade process is not needed for this due to secret defaults are automatically reconciled during the usual reconcile process.
* This is an operator-only functionality: On templates no from address field can be set on the secret nor the environment variable is set in any way (this is also the reason in System options component we still define the smtp from address field as a pointer, to avoid setting it into the secret).

@eguzki let me know about the new changes